### ISSUE TITLE
[flutter_markdown] Add documentation on how to build accessible UIs

### ISSUE THREAD

**Issue by kputh** [flutter/flutter#127485]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Accessibility is an important feature for many apps. Also, there are laws mandating a baseline for accessibility for certain products, e.g. the European Accessibility Act. The _flutter_markdown_ package should have documentation on how accessible resulting UIs are, and how to tweak/improve the result.

### Proposal

The documentation of _flutter_markdown_ should address topics like

* How accessible is the Markdown widget?
* How do you improve accessibility when using the Markdown widget?
  * markdown with links and screen readers
  * markdown with images and screen readers (alt-text)

**Comment by kputh** [comment]
Some context: a colleague of mine mentioned that screen readers omit links in texts rendered with this package. I found no accessibility-related documentation or issues for this package, and wanted to bring the topic (accessibility) to you attention fast. I suggest that you decide how accessible you want the package's widgets to be, and maybe integrate some accessibility testing into your development process. Expect a bug issue regarding the omitted links at a later time.

**Comment by Khushal747** [comment]
I am interested to work on this issue please assign me this task as a SSoC23 contributor

**Comment by stuartmorgan** [comment]
> The _flutter_markdown_ package should have documentation on how accessible resulting UIs are

@kputh Can you elaborate on why you want special documentation about accessibility? Generally we just consider accessibility problems to be bugs, where the solution is to fix them rather than document them.

**Comment by kputh** [comment]
@stuartmorgan I think having some commitment to accessibility in your documentation is a good start to integrating accessibility testing into a development workflow. And I recommend you to. Because I know there is an easy-to-spot issue in this package. :)

Also, currently _flutter_markdown_ has no mention of accessibility in its documentation or issues. As a developer, if you don't assume accessibility is a priority for any package, you're plainly guessing when you decide whether _flutter_markdown_ is a good fit for an app with accessibility requirements, or not.

**Comment by stuartmorgan** [comment]
> I think having some commitment to accessibility in your documentation is a good start to integrating accessibility testing into a development workflow.

Documentation and testing are orthogonal; the way to increase accessibility testing is to add accessibility testing.

> Because I know there is an easy-to-spot issue in this package. :)

And as I suggested above, that should be filed as an issue, and fixed. Adding text to the README would not fix that issue, or prevent future issues.

(Just as the commitment to accessibility [here](https://docs.flutter.dev/accessibility-and-localization/accessibility) did not.)

**Comment by kputh** [comment]
I see your point about accessibility testing and documentation.

How about adding something simple like "This package generates an internal Semantics tree for the provided markdown."? This way developers will know that the don't need a separate solution for screen readers.

**Comment by stuartmorgan** [comment]
> How about adding something simple like "This package generates an internal Semantics tree for the provided markdown."?

As far as I can tell, `flutter_markdown` does not built a `Semantics` tree, it just builds a UI from standard framework widgets, and the accessibility of those widgets is purely a function of the framework widgets themselves rather than anything the package is doing. So unless I'm missing something in the code, that would be a misleading thing to say.

> This way developers will know that the don't need a separate solution for screen readers.

I still don't understand why you believe this needs to be documented on a per-package basis. Widgets should be accessible—I've linked you to existing documentation saying that already for the Flutter project, and this is part of the Flutter project—and if they aren't it's a bug. I don't know why anyone would assume otherwise.

We don't say "This package generates an internal render tree for the provided markdown" so that developers know that they don't need a separate solution for rendering to screen. Why is the semantics tree different?
